unity-client/Assets/Scripts/MainScripts/DCL/Components/BaseComponent.cs

```diff
@@ -12,10 +12,14 @@ public interface IComponent : ICleanable
         Coroutine routine { get; }
         string componentName { get; }
         void UpdateFromJSON(string json);
-        IEnumerator ApplyChanges(string newJson);
+        void UpdateFromObject(BaseModel model);
+        IEnumerator ApplyChanges(BaseModel model);
         void RaiseOnAppliedChanges();
         ComponentUpdateHandler CreateUpdateHandler();
         bool IsValid();
+        BaseModel GetModel();
+        void SetModel(BaseModel model);
```

Comment:
I'm just wondering if `SetModel` is useful now, as we have `UpdateFromObject` that should serve the same purpose. Maybe it's confusing to have both from an interface design perspective.